pkg/kubectl/cmd/patch.go

```
@@ -150,17 +150,14 @@ func RunPatch(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 		}
 
 		helper := resource.NewHelper(client, mapping)
-		patchedObject, err := helper.Patch(namespace, name, patchType, patchBytes)
+		_, err = helper.Patch(namespace, name, patchType, patchBytes)
 		if err != nil {
 			return err
 		}
 		if cmdutil.ShouldRecord(cmd, info) {
-			if err := cmdutil.RecordChangeCause(patchedObject, f.Command()); err == nil {
-				// don't return an error on failure.  The patch itself succeeded, its only the hint for that change that failed
-				// don't bother checking for failures of this replace, because a failure to indicate the hint doesn't fail the command
-				// also, don't force the replacement.  If the replacement fails on a resourceVersion conflict, then it means this
-				// record hint is likely to be invalid anyway, so avoid the bad hint
-				resource.NewHelper(client, mapping).Replace(namespace, name, false, patchedObject)
+			patch, err := cmdutil.ChangeResourcePatch(info, f.Command())
+			if err == nil {
+				helper.Patch(info.Namespace, info.Name, api.StrategicMergePatchType, patch)
```

Add a comment about suppressing the failure and why its good to do that.